### PR TITLE
Restrict leading zero before separator for decimal integers

### PR DESCRIPTION
**Pull request by MaxGraey**
ban literals with leading zero before separator like:
```ts
0_0
0_123
```

- [x] I've read the contributing guidelines

### CODE REVIEW

File: src/tokenizer.ts [hunk 1]
```diff
@@ -1362,6 +1362,11 @@ export class Tokenizer extends DiagnosticEmitter {
               : DiagnosticCode.Multiple_consecutive_numeric_separators_are_not_permitted,
             this.range(pos)
           );
+        } else if (pos - 1 == start && text.charCodeAt(pos - 1) == CharCode._0) {
```
Comment (dcodeIO) [inline src/tokenizer.ts:1365]: Is `00_something` valid?

Comment (MaxGraey) [reply src/tokenizer.ts:1365]: no. It handled before as `ERROR 1121: "Octal literals are not allowed in strict mode.`

Comment (dcodeIO) [reply src/tokenizer.ts:1367]: I see. Can we add a test case just to be sure?

Comment (MaxGraey) [reply src/tokenizer.ts:1367]: sure!

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**